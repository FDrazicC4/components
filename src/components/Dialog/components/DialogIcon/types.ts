import { Props } from '../../../Icon/types';

export { Props };
