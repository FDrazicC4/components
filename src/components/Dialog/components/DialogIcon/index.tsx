import React, { ReactElement } from 'react';
import clsx from 'clsx';
import Icon from '../../../Icon';
import { Props } from './types';


function DialogIcon(props: Props): ReactElement {
    const { className, ...otherProps } = props;
    const classes = clsx(
        'brew-Dialog__icon',
        className,
    );

    return (
        <Icon className={classes} {...otherProps} />
    );
}

export default DialogIcon;
