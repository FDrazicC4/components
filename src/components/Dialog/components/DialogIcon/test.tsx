import React from 'react';
import renderer from 'react-test-renderer';
import DialogIcon from './index';
import { shallow } from 'enzyme';

describe('DialogIcon', () => {
    it('matches the current snapshot', () => {
        const dialogIcon = renderer.create(<DialogIcon>icon</DialogIcon>).toJSON();
        expect(dialogIcon).toMatchSnapshot();
    });

    it('mounts with the right classes', () => {
        expect(shallow(<DialogIcon>icon</DialogIcon>).hasClass('brew-Dialog__icon')).toBe(true);
    });

    it('passes through class names correctly', () => {
        const testClassName = `class-${Math.floor(Math.random() * 1000).toString()}`;
        expect(shallow(<DialogIcon className={testClassName}>icon</DialogIcon>).hasClass(testClassName)).toBe(true);
    });

    it('accepts a children prop correctly', () => {
        const children = 'children';
        expect(shallow(<DialogIcon>{children}</DialogIcon>).contains(children)).toBe(true);
    });
});
