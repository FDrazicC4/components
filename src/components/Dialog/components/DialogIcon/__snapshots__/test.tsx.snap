// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogIcon matches the current snapshot 1`] = `
<span
  className="wrapper color--inherit size--inherit brew-Dialog__icon"
>
  <svg
    height="100%"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    ‚òê
  </svg>
</span>
`;
